### ISSUE TITLE
`destructure-keys` command can break code with non-unique keyword names

### ISSUE THREAD

**Issue by escherize** [clojure-lsp/clojure-lsp#1554]
**Describe the bug**
Given a function argument that is a map with namespaced keywords, where the keyword names are not unique, the `destructure-keys` command is busted.

**To Reproduce**

Given the following function in my editor:
```
(defn foo [bar]
  (str "a: " (:a bar) " x/a: " (:x/a bar)))

(foo {:a "A" :x/a "X"})
;; => "a: X x/a: X"
```

All is well, until I run the `destructure-keys` command:

```
(defn foo [{:keys [a x/a]}]
  (str "a: " a " x/a: " a))

(foo {:a "A" :x/a "X"})
;; => "a: X x/a: X"
```

**Expected behavior**
I expected the destructure-keys command to not alter the output of the function it is called on.

<details>
 <summary><b>clojure-lsp output</b></summary>
<pre>
(It works correctly, there's an issue with the implementation).
</pre>
</details>

**Comment by ericdallo** [comment]
FYI @mainej in case you are interested

**Comment by mainej** [comment]
@escherize ... I see a few way we could handle this case.
 1. `clojure-lsp` should recognize that the refactoring will cause variable shadowing and refuse to do it.
 2. `clojure-lsp` should keep the current behavior. That is, it should do the refactoring and add an `unused binding` lint to `a`, which is shadowed by `x/a`.
 3. Some hybrid where `clojure-lsp` randomly chooses to destructure either `:a` or `:x/a` and leaves the other one as is.

I'm not in favor of option 3... I find it annoying when automated tools make random decisions that I can't influence.

As for options 1 and 2, I'm divided. Let's assume that you'd like to get to this state, which avoids the shadowing:

```clojure
(defn foo [{:keys [x/a] :as bar}]
  (str "a: " (:a bar) " x/a: " a))
```

Option 1, aborting, wouldn't help you at all. You'd have to do the refactoring manually, typing `{:keys [x/a] :as bar}` in place of `bar`, and `a` in place of `(:x/a bar)`.

Option 2, continuing, gives you options. You could undo, which essentially puts you in the same position as Option 1, where you have to do the entire refactoring manually. Or you could leave the automated refactoring as is and manually fix the shadowing by adding `:as bar` and reverting the first `a` to `(:a bar)`. Or you could run the automated refactoring, copy a few bits that would be harder to type, like `{:keys ...}`, undo, paste in what you had copied, and adjust from there.

I lean toward Option 2 because you have more options. In cases where there are a lot of keys to destructure, fixing a little bit of shadowing can be much faster than doing the whole refactoring manually.

On the other hand, I pay close attention to the warnings emitted by clojure-lsp (or really, under the hood, by clj-kondo), and so I would pretty quickly notice the `unused binding` lint indicating the shadowing. For users that don't pay as much attention to lint, this shadowing is more likely to go unnoticed.

There are two sides to the debate—"clojure-lsp shouldn't break things", and, "clojure-lsp is a sharp tool, and should be used with respect and a certain amount of caution". @ericdallo perhaps you have an opinion on which is better in this case.

**Comment by ericdallo** [comment]
Agreed, but I think there is a option 4 that may fit most cases and try to make things clear to user: ask if user wants to procceed with the refactoring (following option 2) or noop (option 1), making clear the message what would result in choosing to proceed, WDYT? we could use the `producer/show-message-request` to control that.

**Comment by hiredman** [comment]
clojure-lsp could also generate non-colliding binding names:
```
(defn foo [{a :a a-1 :x/a}]
  (str "a: " a " x/a: " a-1))
```
